### ISSUE TITLE
boto_vpc.accept_vpc_peering_connection wait a object 

### ISSUE THREAD

**Issue by arthtux** [saltstack/salt#41286]
### Description of Issue

After create a request for peering connection the state boto_vpc.accept_vpc_peering_connection return a error because the function in module boto_vpc return a bool and the state accept_vpc_peering_connection wait a object. 

```
          ID: Accept the vpc peering connection bastion openfleet
    Function: boto_vpc.accept_vpc_peering_connection
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1746, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1703, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/boto_vpc.py", line 1497, in accept_vpc_peering_connection
                  if not pending['exist']:
              TypeError: 'bool' object has no attribute '__getitem__'
     Started: 20:38:34.568862
```

### Setup

```
Accept the vpc peering connection bastion {{ name }}:
  boto_vpc.accept_vpc_peering_connection:
    - profile: {{ profile }}
    - conn_name: bastion-{{ name }}
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.3

Dependency Versions:
           cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 2.5.1
           gitdb: 0.5.4
           gitpython: 0.3.2 RC1
           ioflo: Not Installed
           Jinja2: 2.7.2
           libgit2: Not Installed
           libnacl: Not Installed
           M2Crypto: Not Installed
           Mako: 0.9.1
           msgpack-pure: Not Installed
           msgpack-python: 0.4.6
           mysql-python: 1.2.3
           pycparser: Not Installed
           pycrypto: 2.6.1
           pygit2: Not Installed
           Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           python-gnupg: Not Installed
           PyYAML: 3.10
           PyZMQ: 14.0.1
           RAET: Not Installed
           smmap: 0.8.2
           timelib: Not Installed
           Tornado: 4.2.1
           ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
           machine: x86_64
           release: 3.13.0-29-generic
           system: Linux
           version: Ubuntu 14.04 trusty
```

**Comment by gtmanfred** [comment]
Thanks for the PR for this.

Looks like that should fix it!

Thanks,
Daniel

**Comment by shaidar** [comment]
Not sure whether the PR fixed it as I'm running into the same error with Salt 2017.7.0. Calling this:
```accept_operations_vpc_peering_connection_with_default:
    boto_vpc.accept_vpc_peering_connection:
      - conn_name: {{ VPC_RESOURCE_SUFFIX }}-default-peer
      - require:
        - boto_vpc: create_operations_vpc_peering_connection_with_default
```

Getting the following error:
``` ID: accept_operations_vpc_peering_connection_with_default
    Function: boto_vpc.accept_vpc_peering_connection
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1837, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1794, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/boto_vpc.py", line 1514, in accept_vpc_peering_connection
                  if not pending['exists']:
              TypeError: 'bool' object has no attribute '__getitem__'
     Started: 15:50:07.755052
    Duration: 18.824 ms
```
Here's my versions-report:
```Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.0
         Python: 2.7.13 (default, Jan 19 2017, 14:48:08)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.1
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-3-amd64
         system: Linux
        version: debian 9.1
```

**Comment by shaidar** [comment]
Looks like the fix didn't make it into 2017.7.0 for some reason, so applied it and things seem to be working.
